docs/index.rst

```diff
@@ -1,7 +1,186 @@
-   contain the root `toctree` directive.
+.. include:: ../README.rst
+
+Installation
+------------
+
+Install ``django-select2``
+
+.. code-block:: python
+
+    python3 -m pip install django-select2
+
+Add ``django_select2`` to your ``INSTALLED_APPS`` in your project settings.
+
+Add ``django_select`` to your urlconf if you use any "Auto" fields.
+
+.. code-block:: python
+
+    from django.urls import include, path
+
+    urlpatterns = [
+        # … other patterns
+        path("select2/", include("django_select2.urls")),
+        # … other patterns
+    ]
+
+Finally make sure you have a persistent cache backend setup
+(not the dummy backen), we will use Redis in this example.
+Make sure you have a Redis server up and running::
+
+    # Debian
+    sudo apt-get install redis-server
+    # macOS
+    brew install redis
+    # install Redis python client
+    python3 -m pip install django-redis
+
+Next, add the cache configuration to your ``settings.py`` as follows:
+
+.. code-block:: python
+
+    CACHES = {
+        # … default cache config and others
+        "select2": {
+            "BACKEND": "django_redis.cache.RedisCache",
+            "LOCATION": "redis://127.0.0.1:6379/2",
+            "OPTIONS": {
+                "CLIENT_CLASS": "django_redis.client.DefaultClient",
+            }
+        }
+    }
+
+    # Tell select2 which cache configuration to use:
+    SELECT2_CACHE_BACKEND = "select2"
+
+
+External Dependencies
+---------------------
+
+-  jQuery is not included in the package since it is
+   expected that in most scenarios this would already be available.
+
+
+Quick Start
+-----------
+
+Here is a quick example to get you started:
+
+First make sure you followed the installation instructions above.
+Once everything is setup, let's start with a simple example.
+
+We have the following model:
+
+.. code-block:: python
+
+    # models.py
+    from django.conf import settings
+    from django.db import models
+
+
+    class Book(models.Model):
+        author = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+        co_authors = models.ManyToManyField(settings.AUTH_USER_MODEL, related_name='co_authors')
+
+
+Replace native Django forms widgets with one of the several ``django_select2.form`` widgets.
+Start by importing them into your ``forms.py``, right next to Django own ones:
+
+.. code-block:: python
+
+    # forms.py
+    from django import forms
+    from django_select2 import forms as s2forms
+
+    from . import models
+
+
+    class AuthorWidget(s2forms.ModelSelect2Widget):
+        search_fields = [
+            "username__icontains",
+            "email__icontains",
+        ]
+
+
+    class CoAuthorWidget(s2forms.ModelSelect2MultipleWidget):
+        search_fields = [
+            "username__icontains",
+            "email__icontains",
+        ]
+
+
+    class BookForm(forms.ModelForm):
+        class Meta:
+            model = models.Book
+            fields = "__all__"
```

Comment:
`__all__` has security implications so let's not advertise it but be explicit here, okay?